fastembed/embedding.py

```diff
@@ -464,7 +527,8 @@ def embed(
 
         if parallel is None or is_small:
             for batch in iter_batch(documents, batch_size):
-                yield from self.model.onnx_embed(batch)
+                embeddings, _ = self.model.onnx_embed(batch)
```

Comment:
Hmm, I'm confused. I believe `FlagEmbedding` should be left untouched since all the changes are in the parent class and JinaAI Embedding class, right? 

Similarly, the list_supported_models rewrite isn't needed and should be removed from all implementations now?

Reply:
`FlagEmbedding` cannot be left entirely untouched unfortunately, unless I am missing something.

Before this PR, the `EmbeddingModel.onnx_embed()` method picks out the first token as form of pooling, and then applies normalization. Baked in with this is the assumption that all subclasses of `Embedding` (that hold an `EmbeddingModel` instance) intend for that behaviour. That assumption is broken by Jina embeddings, which requires mean pooling before the normalization.
And mean pooling cannot be applied after this, since the existing implementation of `EmbeddingModel.onnx_embed()` "throws away" the tokens needed for that.

Therefore, the implementation of `EmbeddingModel.onnx_embed()` needs two small modifications:
1. It delegates pooling and normalization to the subclasses of `Embedding`
2. It returns the tokenizer's attentions mask. Otherwise, without access to the attention mask, pooling schemes such as mean pooling cannot be implemented on the `Embedding` level.

This requires `FlagEmbedding` to adjust to those changes.
Just like `JinaEmbedding`, it now implements its own pooling scheme (just picking out the first token). The attention mask is not required for this, so it can be ignored when returned by `EmbeddingModel.onnx_embed()`.

Reply:
As for the `list_supported_models()` rewrite, yes, I can remove that. But then `JinaEmbedding.list_supported_models()` would return a bunch of models that are actually not supported by the `JinaEmbedding` class.

Reply:
I see your point. Looks like we've to figure out a way to handle normalize, attention and pooling steps separately for each embedding implementation. At the moment, what you've proposed kinda works. 

Let me think about this + test your PR and then we're good to go and merge this.